tests/core.test.js

```diff
@@ -0,0 +1,183 @@
+    const returnedFunction = core.mapToAwsMethod("mockMethod");
+
+    returnedFunction(mockClient);
+    expect(mockClient.mockMethod).toHaveBeenCalled();
+    expect(mockMethodPromise).toHaveBeenCalled();
+  });
+
+  test("returns a function that throws if no such method has been found", () => {
+    const returnedFunction = core.mapToAwsMethod("nonExistentMethod");
+    expect(() => returnedFunction(mockClient)).toThrowError();
+  });
+});
+
+describe("bootstrap", () => {
+
+  test("returns a single object containing all of the functions passed as arguments", () => {
+    core.__set__("wrapPluginMethod", jest.fn());
+    core.__set__("wrapAutocompleteFunction", jest.fn());
+
+    const bootstrappedObject = core.bootstrap({}, mockPluginMethods, mockAutocompleteFunctions);
+    expect(_.keys(bootstrappedObject)).toHaveLength(_.entries(mockPluginMethods).length + _.entries(mockAutocompleteFunctions).length);
+    expect(_.keys(bootstrappedObject)).toStrictEqual([..._.keys(mockPluginMethods), ..._.keys(mockAutocompleteFunctions)]);
+  });
+
+  test("properly wraps plugin methods by returning the function that calls the original plugin method with correct arguments", () => {
+    const mockMethodParameters = {
+      action: mockAction,
+      settings: mockSettings
+    };
+    core.__set__("getServiceInstance", jest.fn().mockImplementation(() => mockClient));
+    core.__set__("helpers.readActionArguments", jest.fn().mockImplementation(() => mockParsedParams));
+    core.__set__("helpers.readRegion", jest.fn().mockImplementation(() => mockRegion));
+
+    const bootstrappedObject = core.bootstrap(mockServiceConstructor, mockPluginMethods, {});
+    bootstrappedObject.method1(mockAction, mockSettings);
+
+    expect(mockPluginMethods.method1).toHaveBeenCalledWith(mockClient, mockParsedParams, mockRegion, mockMethodParameters);
+  });
+
+  test("properly wraps autocomplete functions by returning the function that calls the original autocomplete function with correct arguments", () => {
+    const mockAutocompleteParameters = {
+      pluginSettings: mockAutocompleteSettings,
+      actionParams: mockAutocompleteParams,
+    };
+    core.__set__("autocomplete.mapAutocompleteFuncParamsToObject", jest.fn().mockImplementation(() => mockParsedParams));
+    core.__set__("getServiceInstance", jest.fn().mockImplementation(() => mockClient));
+    core.__set__("helpers.readRegion", jest.fn().mockImplementation(() => mockRegion));
+
+    const bootstrappedObject = core.bootstrap(mockServiceConstructor, {}, mockAutocompleteFunctions);
+    bootstrappedObject.auto1(mockQuery, mockAutocompleteSettings, mockAutocompleteParams);
+
+    expect(mockAutocompleteFunctions.auto1)
+      .toHaveBeenCalledWith(mockQuery, mockParsedParams, mockClient, mockRegion, mockAutocompleteParameters);
+  });
+});
+
+describe("getServiceInstance", () => {
+  test("calls provided service constructor per method call and returns its the result", async () => {
```

Comment:
It doesn't call `getServiceInstance`, it check if it's been called.
What about `checks if the "service is created"/"getServiceInstance is called" for every plugin method`?